test/fn-api-tests/exec_test.go

```diff
@@ -66,6 +67,47 @@ func TestCanCallfunction(t *testing.T) {
 	DeleteApp(t, s.Context, s.Client, s.AppName)
 }
 
+func TestCanCallALotOfHotFunctions(t *testing.T) {
+	s := SetupDefaultSuite()
+	CreateApp(t, s.Context, s.Client, s.AppName, map[string]string{})
+	CreateRoute(t, s.Context, s.Client, s.AppName, s.RoutePath,
+		"hhexo/hot", "sync", "http", s.Timeout, int32(5),
+		s.RouteConfig, s.RouteHeaders)
+
+	u := url.URL{
+		Scheme: "http",
+		Host:   Host(),
+	}
+	u.Path = path.Join(u.Path, "r", s.AppName, s.RoutePath)
+
+	errs := make(chan error)
+	numCalls := 50
```

Comment:
we'll probably want to be a little more precise than this, specifically wrt how much memory do we have & how long do we expect this to take, since depending on the size of the instance in which this test suite is run this can change dramatically. for example, a dev using 1GB of RAM for their vm to dev may not be able to get this passing, even if they didn't 'break' this behavior -- one option is to skip under a certain size. since we're running inside of a CI container we'll likely have to work to make this test not flaky. 

wrt breaking other tests (as mentioned), one option is to move this into the agent or server tests, where you could call `agent.Close()` after the test is over to shut down all the hot containers.